proposals/22-04-localize-command.md

```diff
@@ -0,0 +1,335 @@
+│   ├── overlay
+│   │   └── kustomization.yaml
+│   ├── localized-overlay
+│   │   ├── kustomization.yaml
+│   │   └── localized-files
+│   │       └── github.com
+│   │           └── kubernetes-sigs
+│   │               └── kustomize
+│   │                   └── v1.0.6
+│   │                       └── examples
+│   │                           └── helloWorld
+│   │                               └── deployment.yaml
+│   └── base
+│       └── kustomization.yaml
+└── new-space
+    └── localized-example            # the new, localized kustomization directory
+        ├── kustomization.yaml
+        ├── base
+        │   ├── kustomization.yaml
+        │   └── localized-files
+        │       └── github.com
+        │           └── kubernetes-sigs
+        │               └── kustomize
+        │                   └── v1.0.6
+        │                       └── examples
+        │                           ├── helloWorld
+        │                           │   └── configMap.yaml
+        │                           └── multibases
+        │                               ├── base
+        │                               │   └── pod.yaml
+        │                               ├── dev
+        │                               ├── kustomization.yaml
+        │                               ├── production
+        │                               └── staging
+        └── overlay
+            ├── kustomization.yaml
+            └── localized-files
+                └── github.com
+                    └── kubernetes-sigs
+                        └── kustomize
+                            └── v1.0.6
+                                └── examples
+                                    └── helloWorld
+                                        └── deployment.yaml
+```
+```
+# new-space/localized-example/overlay/kustomization.yaml
+resources:
+  - ../base
+  - ./localized-files/github.com/kubernetes-sigs/kustomize/examples/helloWorld/deployment.yaml
+```
+```
+# new-space/localized-example/base/kustomization.yaml
+resources:
+  - ./localized-files/github.com/kubernetes-sigs/kustomize/examples/multibases
+  - ./localized-files/github.com/kubernetes-sigs/kustomize/examples/helloWorld/configMap.yaml
+```
+
+Now, I upload `new-space/localized-example` from my local setup to my company’s internal package management site. I
+change the commands in my CI/CD pipeline to pull `localized-example` before running `kustomize build localized-example`,
+and the command executes successfully!
+
+### Risks and Mitigations
+N/A
+
+### Dependencies
+N/A
+
+### Scalability
+
+A chain of remote kustomization directories in which the current kustomization file references the next remote
+kustomization directories could create a `newDir` with deeply nested `localized-files`. This directory structure would
+impede users’ navigation of `newDir`. However, this scenario should be unlikely as most kustomizations only consist of a
```

Comment:
Might large chains also have performance issues? How about references to large remote repos?

Reply:
Yes, `localize` will take longer for large chains and remote repos, but I believe that is unavoidable. `build` also takes longer for these use cases, right? 

Should I add a sentence about the inevitability of this, or is there a known way to address this? Parallel workers?

Reply:
> `build` also takes longer for these use cases, right?

This is true, we have lots of issues regarding this (e.g. https://github.com/kubernetes-sigs/kustomize/issues/2460)

> Should I add a sentence about the inevitability of this, or is there a known way to address this? Parallel workers?

We don't currently have any parallelism in the kustomize code, and there are already parts of it that we know are not threadsafe so I'm hesitant to recommend this. IMO we can acknowledge that there are these known cases that will have performance problems, and that any solution for them should also help the performance of `build`. WDYT @KnVerey?

Reply:
Yep, ok to simply acknowledge here